src/Resources/ResourceManager/Implementation/ResourceGroups/ExportAzureResourceGroupCmdlet.cs

```diff
@@ -93,63 +96,90 @@ public class ExportAzureResourceGroupCmdlet : ResourceManagerCmdletBaseWithApiVe
+        [CmdletParameterBreakingChange("ApiVersion", ChangeDescription = "Parameter is being deprecated without being replaced")]
+        [Parameter(Mandatory = false, HelpMessage = "When set, indicates the version of the resource provider API to use. If not specified, the API version is automatically determined as the latest available.")]
+        [ValidateNotNullOrEmpty]
+        public override string ApiVersion { get; set; }
+
         /// <summary>
         /// Executes the cmdlet.
         /// </summary>
         protected override void OnProcessRecord()
         {
             base.OnProcessRecord();
+            String contents;
+
             if (ShouldProcess(ResourceGroupName, VerbsData.Export))
             {
 
                 var resourceGroupId = this.GetResourceGroupId();
 
-                var apiVersion = this.ApiVersion ?? DefaultApiVersion;
-
-                var parameters = new ExportTemplateParameters
+                if (this.ApiVersion is null)
                 {
-                    Resources = this.GetResourcesFilter(resourceGroupId: resourceGroupId),
-                    Options = this.GetExportOptions(),
-                };
-
-                var operationResult = this.GetResourcesClient()
-                    .InvokeActionOnResource<JObject>(
-                        resourceId: resourceGroupId,
-                        action: Constants.ExportTemplate,
-                        parameters: parameters.ToJToken(),
-                        apiVersion: apiVersion,
-                        cancellationToken: this.CancellationToken.Value)
-                    .Result;
-
-                var managementUri = this.GetResourcesClient()
-                    .GetResourceManagementRequestUri(
-                        resourceId: resourceGroupId,
-                        apiVersion: apiVersion,
-                        action: Constants.ExportTemplate);
-
-                var activity = string.Format("POST {0}", managementUri.PathAndQuery);
-                var resultString = this.GetLongRunningOperationTracker(activityName: activity,
-                    isResourceCreateOrUpdate: false)
-                    .WaitOnOperation(operationResult: operationResult);
-
-                var template = JToken.FromObject(JObject.Parse(resultString)["template"]);
-
-                if (JObject.Parse(resultString)["error"] != null)
+                    var parameters = new Management.ResourceManager.Models.ExportTemplateRequest
+                    {
+                        Resources = this.GetResourcesFilter(resourceGroupId: resourceGroupId),
+                        Options = this.GetExportOptions(),
+                    };
+
+                    var exportedTemplate = ResourceManagerSdkClient.ExportResourceGroup(ResourceGroupName, parameters);
+
+                    var template = exportedTemplate.Template;
+                    contents = template.ToString();
+
+                    var error = exportedTemplate.Error;
```

Comment:
>error [](start = 24, length = 5)

We should somehow output this error. It looks like REST API based call below outputs the error as warning (WriteWarning()), it could also work here to output the error as warning.